pyangbind/lib/yangtypes.py

```diff
@@ -30,61 +30,157 @@
-    "int",
-    "global",
-    "decimal",
-    "float",
+# reserved names in Python, such as being builtin types and keywords.
+# This encompasses both py2 and py3, and generated using the following:
+#
+#     import six
+#     import string
+#     import keyword
+#     if six.PY3:
+#       import builtins
+#     else:
+#       import __builtin__ as builtins
+#
+#     kw = [x.lower() for x in keyword.kwlist]
+#     bi = [x for x in dir(builtins) if not x.startswith('_') and not x[0] in string.ascii_uppercase]
+#     set(kw + bi)
+#
+# And then add the resulting set for each python version plus the pyangbind names.
+reserved_name = {
+    "abs",
+    "all",
+    "and",
+    "any",
+    "apply",
     "as",
-    "if",
-    "else",
-    "elif",
-    "map",
-    "set",
+    "ascii",
+    "assert",
+    "basestring",
+    "bin",
+    "bool",
+    "break",
+    "buffer",
+    "bytearray",
+    "bytes",
+    "callable",
+    "chr",
     "class",
+    "classmethod",
+    "cmp",
+    "coerce",
+    "compile",
+    "complex",
+    "continue",
+    "copyright",
+    "credits",
+    "def",
+    "del",
+    "delattr",
+    "dict",
+    "dir",
+    "divmod",
+    "elif",
+    "else",
+    "enumerate",
+    "eval",
+    "except",
+    "exec",
+    "execfile",
+    "exit",
+    "false",
+    "file",
+    "filter",
+    "finally",
+    "float",
+    "for",
+    "format",
     "from",
+    "frozenset",
+    "getattr",
+    "global",
+    "globals",
+    "hasattr",
+    "hash",
+    "help",
+    "hex",
+    "id",
+    "if",
     "import",
-    "pass",
-    "return",
+    "in",
+    "input",
+    "int",
+    "intern",
     "is",
-    "exec",
-    "pop",
-    "insert",
-    "remove",
-    "add",
-    "delete",
-    "local",
-    "get",
-    "default",
-    "yang_name",
-    "def",
+    "isinstance",
+    "issubclass",
+    "iter",
+    "lambda",
+    "len",
+    "license",
+    "list",
+    "locals",
+    "long",
+    "map",
+    "max",
+    "memoryview",
+    "min",
+    "next",
+    "none",
+    "nonlocal",
+    "not",
+    "object",
+    "oct",
+    "open",
+    "or",
+    "ord",
+    "pass",
+    "pow",
     "print",
-    "del",
-    "break",
-    "continue",
+    "property",
+    "quit",
     "raise",
-    "in",
-    "assert",
-    "while",
-    "for",
+    "range",
+    "raw_input",
+    "reduce",
+    "reload",
+    "repr",
+    "return",
+    "reversed",
+    "round",
+    "set",
+    "setattr",
+    "slice",
+    "sorted",
+    "staticmethod",
+    "str",
+    "sum",
+    "super",
+    "true",
     "try",
-    "finally",
+    "tuple",
+    "type",
+    "unichr",
+    "unicode",
+    "vars",
+    "while",
     "with",
-    "except",
-    "lambda",
-    "or",
-    "and",
-    "not",
+    "xrange",
     "yield",
-    "property",
-    "min",
-    "max",
-]
+    "zip",
+    # additional names for pyangbind API etc
+    "add",
+    "decimal",
+    "default",
+    "delete",
+    "get",
+    "insert",
+    "local",
+    "pop",
+    "remove",
+    "yang_name",
+}
```

Comment:
This is a long enough list, I think it would be worth moving out into a separate file.

Reply:
Also, I think we should keep this as a list instead of a set. Since this isn't expected to change in the code, we can ensure uniqueness ourselves.